### PR TITLE
Add selectors to ha-form

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

Add support for selectors in ha-form.

Goal is to eventually remove all other types and be selector only.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by zsarnett — DISMISSED** [review]

**Review by bramkragten — DISMISSED** [review]

File: src/components/ha-form/types.ts [hunk 1]
```diff
@@ -9,14 +10,24 @@ export type HaFormSchema =
...
-  | HaFormTimeSchema;
+  | HaFormTimeSchema
+  | HaFormSelector;
 
 export interface HaFormBaseSchema {
   name: string;
+  // This value is applied if no data is submitted for this field
   default?: HaFormData;
   required?: boolean;
   optional?: boolean;
```
Comment (bramkragten) [inline src/components/ha-form/types.ts:21]: We should decided which one to use by default, maybe drop 1 from the type

Comment (bramkragten) [reply src/components/ha-form/types.ts:21]: I would say `required`

Comment (balloob) [reply src/components/ha-form/types.ts:21]: Sure, but let's do that in a future PR. We will need to check what is used where. 

For the backend, both required and optional fields can have a default, which is filled in when key is omitted. So not so required or optional…

Comment (balloob) [reply src/components/ha-form/types.ts:21]: I agree with keeping `required`. Let's update it in another PR. It means rewriting all checks for `optional` to be `!required`.

Comment (balloob) [reply src/components/ha-form/types.ts:21]: https://github.com/home-assistant/frontend/pull/11538

**Comment by balloob** [comment]
Don't understand the error

image

**Comment by balloob** [comment]
I bet this has something to do with icons being imported by the supervisor now. I bet @bramkragten  knows the answer when he wakes up

**Comment by bramkragten** [comment]
Yep, that's the issue

**Comment by balloob** [comment]
selector-entity -> ha-entity-picker -> state-badge -> ha-state-icon -> ha-icon -> 💣

**Comment by balloob** [comment]
I've fixed it now by blocking `ha-icon` from being loaded on supervisor builds. It wasn't used since it broke the build. Not sure if this is ok or we should find a solution where we break up selector in a "without icon" version safe for the supervisor and a "with icon" version for HA?

File: gallery/src/pages/components/ha-form.ts [hunk 2]
```diff
@@ -14,6 +20,44 @@ const SCHEMAS: {
...
+      { name: "entity", selector: { entity: {} } },
+      { name: "device", selector: { device: {} } },
+      { name: "area", selector: { area: {} } },
+      { name: "target", selector: { target: {} } },
+      { name: "number", selector: { number: { min: 0, max: 10 } } },
+      { name: "boolean", selector: { boolean: {} } },
+      { name: "time", selector: { time: {} } },
+      { name: "action", selector: { action: {} } },
+      { name: "text", selector: { text: { multiline: false } } },
+      { name: "text_multiline", selector: { text: { multiline: true } } },
```
Comment (frenck) [inline gallery/src/pages/components/ha-form.ts:51]: Forgive me for my ignorance/lack of knowledge on this part.

Can't we handle the selector options?
As in, would be nice if multiline was not defined by the schema name.

The same issue also returns in, for example, the entity selector. Would be nice if we could pass in data from the flow limiting, e.g., the domain you could select.

**Comment by bramkragten** [comment]
I think this is ok. `ha-icon` should never be used in hassio. Only downside is that we no longer have an indication that there is a `ha-icon` in the build.

**Review by bramkragten — APPROVED** [review]

File: gallery/src/pages/components/ha-form.ts (hunk 2 above)
Comment (bramkragten) [reply gallery/src/pages/components/ha-form.ts:51]: That is the case, the name is just for the demo. These are the same selectors that are used in blueprints and service calls.

**Pull request merged**